src/main/kotlin/com/geanik/cvgenerator/service/model/AbstractTemplate.kt

```diff
@@ -0,0 +1,22 @@
+package com.geanik.cvgenerator.service.model
+
+import java.io.ByteArrayOutputStream
+import java.util.zip.ZipEntry
+import java.util.zip.ZipOutputStream
+
+abstract class AbstractTemplate : Template {
+    protected fun writeZip(fileNames: Iterable<String>, fileContents: Iterable<ByteArray>): ByteArray {
+        val zipBos = ByteArrayOutputStream()
+        val zip = ZipOutputStream(zipBos)
+        zip.use {
+            for (content in fileContents) {
+                val entry = ZipEntry("index.html")
```

Comment:
should be more "generic"

not all templates will contain only 1 file called "index.html"

maybe create abstract function "createFiles" or so that all specific classes override and that is then called in the abstract class